pepper-apis/ddp-common/src/main/java/org/broadinstitute/ddp/db/TransactionWrapper.java

```diff
@@ -168,21 +169,19 @@ public static synchronized void init(DbConfiguration... dbConfigs) {
         }
 
         for (DbConfiguration dbConfig : dbConfigs) {
-            if (gTxnWrapper.containsKey(dbConfig.getDb())) {
-                TransactionWrapper transactionWrapper = gTxnWrapper.get(dbConfig.getDb());
-                int maxConnections = dbConfig.getMaxConnections();
-                String dbUrl = dbConfig.getDbUrl();
+            TransactionWrapper.DB db = dbConfig.getDb();
+            int maxConnections = dbConfig.getMaxConnections();
+            String dbUrl = dbConfig.getDbUrl();
+
+            if (gTxnWrapper.containsKey(db)) {
+                TransactionWrapper transactionWrapper = gTxnWrapper.get(db);
                 if (transactionWrapper.maxConnections != maxConnections || !transactionWrapper.dbUrl.equals(dbUrl)) {
-                    throw new RuntimeException("init() has already been called with "
-                            + transactionWrapper.maxConnections + " and "
-                            + transactionWrapper.dbUrl + "; " + "you cannot re-initialize "
-                            + "it with different params " + maxConnections
-                            + " and " + dbUrl);
+                    throw new DDPInternalError("init() has already been called with different parameters: DB=" + db.name());
                 } else {
-                    log.warn("TransactionWrapper has already been initialized.");
+                    log.warn("TransactionWrapper has already been initialized for {}", db.name());
                 }
             }
-            gTxnWrapper.put(dbConfig.getDb(), new TransactionWrapper(dbConfig.getMaxConnections(), dbConfig.getDbUrl(), dbConfig.getDb()));
+            gTxnWrapper.put(db, new TransactionWrapper(maxConnections, dbUrl, db));
```

Comment:
@denniscunningham 
Taking another look at this code (actually existing code though).
We are calling this new TransactionWrapper(maxConnections, dbUrl, db) even if the TransactionWrapper is initialized (we are logging a warning) but still creating DataSources. 
Wondering if the code is called again during the init and are we creating DataSources (connections) again.

Reply:
Whenever we restart Housekeeping I see connections created twice . second set within 10seconds of first connections creation. 
<img width="1702" alt="Screenshot 2023-08-17 at 1 41 34 PM" src="https://github.com/broadinstitute/ddp-study-server/assets/26553609/8d4d2ceb-b02e-4487-aeff-d9c6815ad066">

Reply:
I think I see what you are seeing: there is an initialized flag that is not handled very well. IOW, the flag is set, and there is a way for callers to check it, but nothing that prevents re-entry to the init() method. Is that what you see?

Yup, we can refactor a bit to clean that up. I'd be surprised that it is actually causing an issue, but it should be cleaner.

Reply:
Looks like we are trying to close existing conns and create new set though
if (dataSource != null) {
            dataSource.close();
}

Reply:
<img width="1103" alt="Screenshot 2023-08-17 at 2 21 00 PM" src="https://github.com/broadinstitute/ddp-study-server/assets/26553609/1e7a8174-72ac-4611-842c-891ff8b7dd8b">

Reply:
Dennis: Please go ahead and do cleanup as part of PEPPER-983